src/sentry/utils/snowflake.py

```diff
@@ -55,3 +94,55 @@ def snowflake_id_generation():
     ID_VALIDATOR.validate(snowflake_id)
 
     return snowflake_id
+
+
+def get_redis_cluster(redis_key: str):
+    return redis.clusters.get("default").get_local_client_for_key(redis_key)
+
+
+def store_snowflake_to_redis(redis_key: str, timestamp: int, used_region_sequence: int) -> None:
+    cluster = get_redis_cluster(redis_key)
+    used_region_sequences = cluster.get(str(timestamp))
+    if used_region_sequences:
+        used_region_sequences = json.loads(used_region_sequences)
+    else:
+        used_region_sequences = []
+    used_region_sequences.append(used_region_sequence)
```

Comment:
I'm not sure, just from eyeballing it, whether that race condition would make it possible to add a duplicate value here. But if it does, it would mess up the `len(used_region_sequences) == (MAX_AVAILABLE_REGION_SEQUENCEs)` logic. It might be worth converting `used_region_sequences` to a `set` after reading it from JSON.